Mathlib/Algebra/BigOperators/Finprod.lean

```diff
@@ -351,6 +351,18 @@ theorem finprod_inv_distrib [DivisionCommMonoid G] (f : α → G) : (∏ᶠ x, (
 #align finprod_inv_distrib finprod_inv_distrib
 #align finsum_neg_distrib finsum_neg_distrib
 
+@[to_additive (attr := simp)]
+theorem finprod_prop (P : Prop) [Decidable P] (b : M) : ∏ᶠ _x : P, b = if P then b else 1 := by
+  rw [finprod, dif_pos (toFinite _), Finset.prod_const, mulSupport]
+  simp only [comp_apply, ne_eq]
+  rcases eq_or_ne b 1 with rfl | hb; simp
+  · simp only [hb, not_false_eq_true, setOf_true, toFinite_toFinset, toFinset_univ]
+    split_ifs with h
+    · have : Unique (P) := uniqueProp h
+      simp
+    · have : IsEmpty P := ⟨h⟩
+      simp
```

Comment:
```suggestion
theorem finprod_prop (P : Prop) [Decidable P] (b : M) : ∏ᶠ _x : P, b = if P then b else 1 :=
  finprod_eq_if
```
works, so this lemma is a duplicate.

Reply:
Though I think that's a bad idea too, because then `∏ᶠ x < 37, f x` becomes some awful if.

Reply:
Awful how? I think thats essentially what you want if you have a goal with a finprod over a prop, I see no other way of clearing it tbh.
Or is your example using implicit binders over x also?

Reply:
Also thanks, idk how I missed that lol

Reply:
Oops, I should have spotted this too!

Reply:
> Or is your example using implicit binders over x also?

Yes, exactly.

Reply:
Yeah this is awkward, I really would like something that applies automatically with simp and norm mum in such a situation, but maybe we just have to wait for better big operator support in norm num

Reply:
Either way I'll close this for now. Sorry for the bother